### PR TITLE
test: avoid starting postgres server in the psql pod

### LINKED ISSUE #1243: E2E: createPsqlClient should not create a full DB
The `createPsqlClient` function in the test utils code is creating a Pod
with a PostgreSQL *server* installed, where the expectation is that it
simply should have the `psql` command installed.

### PR DESCRIPTION
**Pull request by hh24k**
Closes #1243

Signed-off-by: Hai He <hai.he@enterprisedb.com>

### CODE REVIEW

**Review by sxd — APPROVED** [review]

File: tests/utils/commons.go [hunk 1]
```diff
@@ -74,11 +74,7 @@ func TestDirectoryEmpty(namespace, podName, directoryPath string) bool {
 func CreateObject(env *TestingEnvironment, object client.Object, opts ...client.CreateOption) error {
 	err := retry.Do(
 		func() error {
-			err := env.Client.Create(env.Ctx, object, opts...)
-			if err != nil {
-				return err
-			}
-			return nil
+			return env.Client.Create(env.Ctx, object, opts...)
```
Comment (jsilvela) [inline tests/utils/commons.go:77]: 👍

File: tests/utils/run.go [hunk 2]
```diff
@@ -58,11 +58,7 @@ func RunUncheckedRetry(command string) (stdout string, stderr string, err error)
 		func() error {
 			cmd := exec.Command(tokens[0], tokens[1:]...) // #nosec G204
 			cmd.Stdout, cmd.Stderr = &outBuffer, &errBuffer
-			err = cmd.Run()
-			if err != nil {
-				return err
-			}
-			return nil
+			return cmd.Run()
```
Comment (jsilvela) [inline tests/utils/run.go:61]: 👍

File: tests/e2e/suite_test.go [hunk 3]
```diff
@@ -148,6 +162,13 @@ func saveOperatorLogs(buf bytes.Buffer, specName string, output io.Writer) {
 	}
 }
 
+// isUpgradeTestSuite tells if current test suite is an upgrade test suite or not basing on
+// the label filter specified in the Ginkgo CLI with the option "--label-filter"
+func isUpgradeTestSuite() bool {
+	suiteConfig, _ := GinkgoConfiguration()
+	return suiteConfig.LabelFilter == tests.LabelUpgrade
+}
+
```
Comment (jsilvela) [inline tests/e2e/suite_test.go:171]: not keen on this. Exceptional flow makes our code more complicated to reason about. I.e. in two months we won't remember that the upgrade run skips client pod creation.
As I understand the issue is with namespace cleanup, let's handle the problem there.
Going to test that hypothesis.

**Review by jsilvela — COMMENTED** [review]
Going to try doing a DeleteNamespaceAndWait, rather than avoiding psqlClientPod creation in the upgrade test...

**Review by jsilvela — APPROVED** [review]

**Pull request merged**